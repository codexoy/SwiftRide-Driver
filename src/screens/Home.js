import React from "react"
import { View, Text, Button, StyleSheet } from "react-native"
import { logout } from "../actions/auth"
import { connect } from "react-redux"

class Home extends React.Component {
	constructor(props) {
		super(props)
		this.logout = this.logout.bind(this)
	}

	logout() {
		this.props.dispatch(
			logout(() => {
				this.props.navigation.navigate("Auth")
			})
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<Text>Hi!</Text>
				<Button onPress={this.logout} title="התנתק" />
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})

export default connect()(Home)
