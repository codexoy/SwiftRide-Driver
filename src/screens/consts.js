import React from "react"
import { Icon } from "react-native-elements"
const options = {
	activeTintColor: "rgb(12,116,244)",
	inactiveTintColor: "black",
	showLabel: false
}

export const tabBarOptions = {
	defaultNavigationOptions: ({ navigation }) => ({
		tabBarIcon: ({ focused, horizontal, tintColor }) => {
			const { routeName } = navigation.state
			let iconName
			let iconType = "material"
			if (routeName === "Home") {
				iconName = "home"
			} else if (routeName === "Notifications") {
				iconName = "notifications"
			} else if (routeName === "Lesson") {
				iconName = "add-circle"
			} else if (routeName === "Students") {
				iconName = "people"
			} else if (routeName == "Schedule") {
				iconName = "date-range"
			} else if (routeName == "Profile") {
				iconName = "person"
			}

			return (
				<Icon
					name={iconName}
					size={25}
					color={tintColor}
					type={iconType}
				/>
			)
		}
	}),
	tabBarOptions: options
}
