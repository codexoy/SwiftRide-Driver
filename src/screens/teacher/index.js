import React from "react"
import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import Home from "./Home"
import Notifications from "./Notifications"
import ChooseDate from "./ChooseDate"
import NewLesson from "./NewLesson"
import Students from "./Students"
import NewStudent from "./NewStudent"
import Schedule from "./Schedule"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"

export default createBottomTabNavigator(
	{
		Home: Home,
		Notifications: Notifications,
		Add: {
			screen: createStackNavigator(
				{ ChooseDate: ChooseDate, NewLesson: NewLesson },
				{
					initialRouteName: "ChooseDate",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "add",
				tabBarLabel: strings("tabs.add"),
				tabBarAccessibilityLabel: strings("tabs.add"),
				tabBarTestID: "NewLessonTab"
			}
		},
		Schedule: Schedule,
		Students: {
			screen: createStackNavigator(
				{ Students: Students, NewStudent: NewStudent },
				{
					mode: "modal",
					initialRouteName: "Students",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "students",
				tabBarLabel: strings("tabs.students"),
				tabBarAccessibilityLabel: strings("tabs.students"),
				tabBarTestID: "StudentsTab"
			}
		}
	},
	{
		defaultNavigationOptions: ({ navigation }) => ({
			tabBarIcon: ({ focused, horizontal, tintColor }) => {
				const { routeName } = navigation.state
				let iconName
				let iconType = "material"
				if (routeName === "Home") {
					iconName = "home"
				} else if (routeName === "Notifications") {
					iconName = "notifications"
				} else if (routeName === "Add") {
					iconName = "add-circle"
				} else if (routeName === "Students") {
					iconName = "people"
				} else if (routeName == "Schedule") {
					iconName = "date-range"
				}

				// You can return any component that you like here!
				return (
					<Icon
						name={iconName}
						size={25}
						color={tintColor}
						type={iconType}
					/>
				)
			}
		}),
		tabBarOptions: {
			activeTintColor: "rgb(12,116,244)",
			inactiveTintColor: "black",
			showLabel: false
		}
	}
)
