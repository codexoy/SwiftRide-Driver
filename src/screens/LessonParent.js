import React, { Fragment } from "react"
import { Alert, View } from "react-native"
import { strings } from "../i18n"
import {
	SHORT_API_DATE_FORMAT,
	GOOGLE_MAPS_QUERY,
	autoCompletePlacesStyle,
	MAIN_PADDING
} from "../consts"
import moment from "moment"
import { GooglePlacesAutocomplete } from "react-native-google-places-autocomplete"
import AlertError from "../components/AlertError"
import { Dropdown } from "react-native-material-dropdown"
import { durationMulOptions } from "./consts"

export default class LessonParent extends AlertError {
	handlePlaceSelection = (name, data) => {
		const place = {
			description: data.description,
			google_id: data.place_id
		}
		this.setState({
			[name + "ListViewDisplayed"]: false,
			[name]: place
		})
		// if clicked on a meetup place and dropoff is empty, fill it up too
		if (
			name == "meetup" &&
			(!this.state.dropoff.hasOwnProperty("description") ||
				this.state.dropoff.description == "")
		) {
			this.setState({ dropoff: place }, () => {
				this.dropoffRef.setAddressText(place.description)
			})
		}
	}

	renderPlaces = () => {
		const places = ["meetup", "dropoff"]
		return places.map((name, index) => {
			return (
				<GooglePlacesAutocomplete
					ref={instance => {
						this[name + "Ref"] = instance
					}}
					key={`autocomplete-${name}`}
					query={GOOGLE_MAPS_QUERY}
					placeholder={strings("teacher.new_lesson." + name)}
					minLength={2}
					autoFocus={false}
					returnKeyType={"default"}
					fetchDetails={false}
					currentLocation={false}
					currentLocationLabel={strings("current_location")}
					nearbyPlacesAPI="GooglePlacesSearch"
					listViewDisplayed={this.state[name + "ListViewDisplayed"]}
					styles={autoCompletePlacesStyle}
					onPress={(data, details = null) => {
						// 'details' is provided when fetchDetails = true
						this.handlePlaceSelection(name, data)
					}}
					getDefaultValue={() => this.state[name].description || ""}
				/>
			)
		})
	}

	deleteConfirm() {
		Alert.alert(strings("are_you_sure"), strings("are_you_sure_delete"), [
			{
				text: strings("cancel"),
				style: "cancel"
			},
			{
				text: strings("ok"),
				onPress: () => {
					this.delete()
				}
			}
		])
	}

	delete = async () => {
		const { lesson } = this.state
		if (!lesson) return
		const resp = await this.props.fetchService.fetch(
			`/lessons/${lesson.id}`,
			{
				method: "DELETE"
			}
		)
		if (resp) {
			Alert.alert(strings("teacher.notifications.lessons_deleted"))
			this.props.navigation.goBack()
		}
	}

	_showDateTimePicker = () => this.setState({ datePickerVisible: true })

	_hideDateTimePicker = () => this.setState({ datePickerVisible: false })

	_handleDatePicked = date => {
		this._hideDateTimePicker()
		this.setState(
			{ date: moment(date).format(SHORT_API_DATE_FORMAT) },
			() => {
				this._getAvailableHours()
			}
		)
	}

	_dropdownChange = (value, index, data) => {
		this.setState(
			{
				duration_mul: value
			},
			() => {
				this._getAvailableHours()
			}
		)
	}

	renderDuration = () => {
		return (
			<Fragment>
				<Dropdown
					value={this.state.duration_mul}
					data={durationMulOptions}
					onChangeText={this._dropdownChange.bind(this)}
					dropdownMargins={{ min: 20, max: 60 }}
					dropdownOffset={{
						top: 0,
						left: 0
					}}
					containerStyle={{
						marginLeft: MAIN_PADDING,
						marginRight: MAIN_PADDING,
						marginTop: 8
					}}
					inputContainerStyle={{
						borderBottomColor: "transparent"
					}}
				/>
			</Fragment>
		)
	}
}
