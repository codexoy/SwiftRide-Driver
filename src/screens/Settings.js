import React, { Fragment } from "react"
import {
	ScrollView,
	Text,
	View,
	StyleSheet,
	TouchableOpacity,
	TouchableHighlight,
	Alert
} from "react-native"
import { connect } from "react-redux"
import { strings, errors } from "../i18n"
import ShadowRect from "../components/ShadowRect"
import {
	MAIN_PADDING,
	floatButtonOnlyStyle,
	NOTIFICATIONS_KEY
} from "../consts"
import PageTitle from "../components/PageTitle"
import { NavigationActions } from "react-navigation"
import { Button, Icon } from "react-native-elements"
import RectInput from "../components/RectInput"
import { logout, setUser } from "../actions/auth"
import { API_ERROR } from "../reducers/consts"
import Storage from "../services/Storage"
import { fetchOrError, popLatestError } from "../actions/utils"

export class Settings extends React.Component {
	constructor(props) {
		// only here for the test suite to work
		super(props)
		this.state = {
			name: this.props.user.name,
			area: this.props.user.area,
			phone: this.props.user.phone,
			password: "",
			price: this.props.user.price,
			duration: this.props.user.lesson_duration,
			notifications: "true"
		}
		this._initNotifications()
	}

	logout = () => {
		this.props.dispatch(
			logout(() => {
				this.props.navigation.navigate("Auth")
			})
		)
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			Alert.alert(strings("errors.title"), errors(error))
		}
	}

	submitTeacherInfo = async () => {
		await this.submit("/teacher/edit_data", {
			price: this.state.price,
			lesson_duration: this.state.duration
		})
	}

	submitInfo = async () => {
		const resp1 = await this.submit("/login/edit_data", {
			name: this.state.name,
			area: this.state.area,
			password: this.state.password,
			phone: this.state.phone
		})
		let resp2 = true
		if (this.props.user.hasOwnProperty("teacher_id")) {
			resp2 = await this.submitTeacherInfo()
		}

		if (resp1) {
			if (resp2) {
				await this.props.dispatch(setUser(resp2.json.data))
			} else {
				await this.props.dispatch(setUser(resp1.json.data))
			}
			Alert.alert(strings("settings.success"))
		}
	}

	submit = async (endpoint, body) => {
		const resp = await this.props.dispatch(
			fetchOrError(endpoint, {
				method: "POST",
				body: JSON.stringify(body)
			})
		)
		return resp
	}

	onChangeText = (param, value) => {
		this.setState({
			[param]: value
		})
	}

	_initNotifications = async () => {
		const notifications = await Storage.getItem(NOTIFICATIONS_KEY)
		this.setState({
			notifications
		})
	}

	toggleNotifications = () => {
		let notifications = "true"
		if (this.state.notifications === "true") {
			notifications = "false"
		}
		this.setState(
			{
				notifications
			},
			async () => {
				await Storage.setItem(NOTIFICATIONS_KEY, notifications)
			}
		)
	}

	render() {
		let workDays, extraForm
		if (this.props.user.hasOwnProperty("teacher_id")) {
			workDays = (
				<TouchableHighlight
					underlayColor="#f8f8f8"
					onPress={() => this.props.navigation.navigate("WorkDays")}
					style={styles.fullWidth}
				>
					<View style={styles.rectInsideView}>
						<Text>{strings("settings.work_hours")}</Text>
					</View>
				</TouchableHighlight>
			)
			extraForm = (
				<Fragment>
					<RectInput
						label={strings("signup.price")}
						iconName="payment"
						value={this.state.price.toString()}
						onChangeText={value =>
							this.onChangeText("price", value)
						}
					/>
					<RectInput
						label={strings("signup.time")}
						iconName="access-time"
						value={this.state.duration.toString()}
						onChangeText={value =>
							this.onChangeText("duration", value)
						}
					/>
				</Fragment>
			)
		}
		return (
			<ScrollView
				keyboardDismissMode="on-drag"
				keyboardShouldPersistTaps="always"
			>
				<View style={styles.container}>
					<PageTitle
						style={styles.title}
						title={strings("settings.title")}
						leftSide={
							<Button
								icon={
									<Icon
										name="ios-close"
										type="ionicon"
										size={36}
									/>
								}
								onPress={() => {
									this.props.navigation.dispatch(
										NavigationActions.back()
									)
								}}
								type="clear"
								style={styles.closeButton}
							/>
						}
					/>
					<Text style={styles.rectTitle}>
						{strings("settings.general")}
					</Text>
					<ShadowRect style={styles.rect}>
						{workDays}
						<TouchableHighlight
							underlayColor="#f8f8f8"
							onPress={this.toggleNotifications.bind(this)}
							style={styles.fullWidth}
						>
							<View style={styles.rectInsideView}>
								<Text style={styles.rightSide}>
									{strings("settings.notifications")}
								</Text>
								<Text style={styles.leftSide}>
									{strings(
										"settings.notifications_" +
											this.state.notifications
									)}
								</Text>
							</View>
						</TouchableHighlight>
						<View style={styles.rectInsideView}>
							<Text>{strings("settings.support")}</Text>
						</View>
					</ShadowRect>
					<Text style={styles.rectTitle}>
						{strings("settings.personal_info")}
					</Text>
					<ShadowRect style={styles.rect}>
						<RectInput
							label={strings("signup.name")}
							iconName="person"
							value={this.state.name}
							onChangeText={value =>
								this.onChangeText("name", value)
							}
						/>
						<RectInput
							label={strings("signup.area")}
							iconName="person-pin"
							value={this.state.area}
							onChangeText={value =>
								this.onChangeText("area", value)
							}
						/>
						<RectInput
							label={strings("signup.phone")}
							iconName="phone"
							value={this.state.phone}
							onChangeText={value =>
								this.onChangeText("phone", value)
							}
						/>
						{extraForm}
						<RectInput
							label={strings("signin.password")}
							iconName="security"
							value={this.state.password}
							onChangeText={value =>
								this.onChangeText("password", value)
							}
							secureTextEntry
						/>
						<TouchableOpacity
							style={styles.button}
							onPress={this.submitInfo.bind(this)}
						>
							<View>
								<Text style={styles.buttonText}>
									{strings("settings.submit")}
								</Text>
							</View>
						</TouchableOpacity>
					</ShadowRect>

					<TouchableOpacity onPress={this.logout.bind(this)}>
						<Text style={styles.logout}>
							{strings("settings.logout")}
						</Text>
					</TouchableOpacity>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		marginTop: 20,
		alignItems: "center"
	},
	closeButton: {
		marginTop: -6
	},
	rect: {
		marginTop: 12,
		marginBottom: 24,
		paddingHorizontal: 0,
		paddingVertical: 0
	},
	rectTitle: {
		fontWeight: "bold",
		alignSelf: "flex-start",
		color: "#5c5959"
	},
	rectInsideView: {
		width: "100%",
		borderBottomWidth: 1,
		borderBottomColor: "#f7f7f7",
		paddingVertical: 6,
		paddingHorizontal: 20,
		paddingVertical: 12,
		flexDirection: "row"
	},
	button: { ...floatButtonOnlyStyle, width: "100%", borderRadius: 0 },
	buttonText: {
		fontWeight: "bold",
		fontSize: 20,
		color: "#fff"
	},
	logout: {
		fontWeight: "bold",
		alignSelf: "center",
		color: "red",
		marginBottom: MAIN_PADDING
	},
	leftSide: { flex: 1, marginLeft: "auto" },
	fullWidth: {
		flex: 1,
		width: "100%"
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		errors: state.errors
	}
}

export default connect(mapStateToProps)(Settings)
