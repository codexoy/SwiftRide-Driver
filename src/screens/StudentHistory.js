import React, { Fragment } from "react"
import {
	View,
	StyleSheet,
	ScrollView,
	Text,
	FlatList,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../i18n"
import PageTitle from "../components/PageTitle"
import { Button, Icon } from "react-native-elements"
import { MAIN_PADDING, colors } from "../consts"
import LessonsLoader from "../components/LessonsLoader"
import EmptyState from "../components/EmptyState"
import LessonPopup from "../components/LessonPopup"
import { getRole } from "../actions/auth"
import Hours from "../components/Hours"

export class StudentHistory extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			appointments: [],
			loading: true,
			visible: []
		}
		this.isTeacher = false
		if (getRole(this.props.user) == "teacher") {
			this.isTeacher = true
		}
		this._getAppointments()
	}

	_getAppointments = async () => {
		const resp = await this.props.fetchService.fetch(
			"/appointments/?is_approved=true&student_id=" +
				this.props.navigation.getParam("student_id"),
			{
				method: "GET"
			}
		)
		console.log(resp.json.data)
		this.setState({
			appointments: resp.json.data,
			loading: false
		})
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	renderItem = ({ item, index }) => {
		const date = item.date
		const meetup = item.meetup_place || strings("not_set")
		const dropoff = item.dropoff_place || strings("not_set")
		const visible = this.state.visible.includes(item.id) ? true : false
		let approved = ""
		if (!item.is_approved) {
			approved = " - " + strings("not_approved")
		}
		let lessonTitle =
			strings("teacher.home.lesson_number") +
			" " +
			item.lesson_number +
			approved

		if (item.type != "lesson") {
			lessonTitle = strings("teacher.new_lesson.types." + item.type)
		}
		return (
			<Fragment>
				<View style={styles.lesson}>
					<TouchableOpacity onPress={() => this.lessonPress(item)}>
						<Text style={styles.lessonTitle}>{lessonTitle}</Text>
						<Hours
							duration={item.duration}
							date={date}
							style={styles.hours}
						/>
						<Text style={styles.places}>
							{strings("teacher.new_lesson.meetup")}: {meetup},{" "}
							{strings("teacher.new_lesson.dropoff")}: {dropoff}
						</Text>
					</TouchableOpacity>
				</View>
				<LessonPopup
					visible={visible}
					item={item}
					onPress={this.lessonPress.bind(this)}
					navigation={this.props.navigation}
					isStudent={!this.isTeacher}
				/>
			</Fragment>
		)
	}

	_renderEmpty = () => {
		if (this.state.loading) return <View />
		return (
			<EmptyState
				image="lessons"
				text={strings("empty_lessons")}
				style={styles.empty}
			/>
		)
	}

	render() {
		let loading
		if (this.state.loading) {
			loading = <LessonsLoader />
		}
		return (
			<ScrollView style={{ flex: 1 }}>
				<View style={styles.container}>
					<View style={styles.headerRow}>
						<Button
							icon={<Icon name="arrow-forward" type="material" />}
							onPress={() => {
								this.props.navigation.goBack()
							}}
							type="clear"
							style={{ marginTop: -6 }}
						/>
						<PageTitle
							style={styles.title}
							title={strings("student_profile.lessons_history")}
						/>
					</View>
					{loading}
					<FlatList
						data={this.state.appointments}
						renderItem={this.renderItem.bind(this)}
						keyExtractor={item => `${item.id}`}
						ListEmptyComponent={this._renderEmpty}
						extraData={this.state.visible}
					/>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginHorizontal: MAIN_PADDING,
		marginVertical: MAIN_PADDING
	},
	headerRow: {
		flexDirection: "row"
	},
	lesson: {
		flex: 1,
		marginTop: 12,
		backgroundColor: "#f4f4f4",
		padding: 8
	},
	lessonTitle: {
		color: colors.blue,
		alignSelf: "flex-start"
	},
	places: {
		fontSize: 14,
		color: "gray",
		marginTop: 4,
		alignSelf: "flex-start",
		textAlign: "left"
	},
	hours: {
		fontSize: 14,
		color: "gray",
		alignSelf: "flex-start"
	}
})

function mapStateToProps(state) {
	return {
		error: state.error,
		fetchService: state.fetchService,
		user: state.user
	}
}

export default connect(mapStateToProps)(StudentHistory)
