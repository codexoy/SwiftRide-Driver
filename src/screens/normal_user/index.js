import React, { Component } from "react"
import { createBottomTabNavigator } from "react-navigation"
import Home from "./Home"
import { tabBarOptions } from "../consts"

export default createBottomTabNavigator(
	{
		Home: Home
	},
	tabBarOptions
)
