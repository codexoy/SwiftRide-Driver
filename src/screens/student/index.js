import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import { tabBarOptions } from "../consts"
import Home from "./Home"
import Notifications from "./Notifications"
import Lesson from "./Lesson"
import Schedule from "./Schedule"
import Profile from "./Profile"
import Settings from "../Settings"
import StudentProfile from "../student/Profile"

export default createBottomTabNavigator(
	{
		Home: createStackNavigator(
			{
				Main: createStackNavigator(
					{ Home, Lesson },
					{
						initialRouteKey: "Home",
						headerMode: "none"
					}
				),
				Settings
			},
			{
				initialRouteName: "Main",
				mode: "modal",
				headerMode: "none",
				navigationOptions: {
					headerVisible: false
				}
			}
		),
		Notifications: createStackNavigator(
			{ Main: Notifications, Lesson },
			{
				initialRouteKey: "Main",
				headerMode: "none"
			}
		),
		AddLesson: Lesson,
		Schedule: createStackNavigator(
			{ Main: Schedule, Lesson },
			{
				initialRouteKey: "Main",
				headerMode: "none"
			}
		),
		StudentProfile: Profile
	},
	tabBarOptions
)
