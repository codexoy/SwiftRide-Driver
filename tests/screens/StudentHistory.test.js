import React from "react"

import { StudentHistory } from "../../src/screens/StudentHistory"

const user = { name: "test", area: "test", phone: "05501111", price: 1000 }
const student = {
	...user,
	student_id: 1,
	balance: 900,
	my_teacher: { user: {} }
}
describe("StudentHistory", () => {
	test("view renders correctly", done => {
		fetch.mockResponseSuccess(
			JSON.stringify({
				data: [
					{
						student,
						dropoff_place: "t",
						meetup_place: "a"
					}
				]
			})
		)
		const wrapper = shallow(
			<StudentHistory
				user={user}
				navigation={{ ...navigation, getParam: param => 1 }}
				dispatch={dispatch}
				fetchService={fetchService}
			/>
		)
		testAsyncComponent(wrapper, done)
	})
})
