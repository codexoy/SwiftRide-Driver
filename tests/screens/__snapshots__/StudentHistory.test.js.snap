// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentHistory view renders correctly 1`] = `
<ScrollViewMock
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 26,
        "marginVertical": 26,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <ForwardRef(Themed.Button)
        icon={
          <Themed.Icon
            name="arrow-forward"
            type="material"
          />
        }
        onPress={[Function]}
        style={
          Object {
            "marginTop": -6,
          }
        }
        type="clear"
      />
      <PageTitle
        title="היסטוריית מפגשים"
      />
    </View>
    <FlatList
      ListEmptyComponent={[Function]}
      data={
        Array [
          Object {
            "dropoff_place": "t",
            "meetup_place": "a",
            "student": Object {
              "area": "test",
              "balance": 900,
              "my_teacher": Object {
                "user": Object {},
              },
              "name": "test",
              "phone": "05501111",
              "price": 1000,
              "student_id": 1,
            },
          },
        ]
      }
      disableVirtualization={false}
      extraData={Array []}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</ScrollViewMock>
`;
